src/cargo/core/resolver/mod.rs

```diff
@@ -933,13 +933,13 @@ fn activation_error(
         };
 
         let mut msg = format!(
-            "no matching version `{}` found for package `{}`\n\
-             location searched: {}\n\
-             versions found: {}\n",
-            dep.version_req(),
+            "failed to select a version for `{}` satisfying the requirement `{}`\n  \
```

Comment:
I found this sentence structure convoluted the first time I looked at this PR, but it has grown on me. I thought about putting the concepts in the other order, but that leads to the existing wording which many people report as confusing.

A more radical thought is to use the cargo.toml syntax:
`failed to select a version satisfying the requirement 'foo = "2"'`
But I do not know if that is a good thought.

Reply:
Oh I like that actually, everyone's quite familiar with it!